Compilation error of dispatch.c

Compiler explorer daily builds are broken due to the compilation error of dispatch.c (link: https://github.com/compiler-explorer/compiler-workflows/actions/runs/12246779057).

```bash
root@a147a4aec370:~/ispc# /root/ispc/build/stage2/bin/clang++ -x c -DMACOS -fPIC --target=x86_64-unknown-unknown -march=core2 -mtune=generic -O2 -emit-llvm builtins/dispatch.c -c -o /root/ispc/build/build-ispc-stage2/src/ispc-stage2-build/bin/../share/ispc/builtins_dispatch_macos.bc -v
clang version 18.1.8 (https://github.com/llvm/llvm-project.git 3b5b5c1ec4a3095ab096dd780e84d7ab81f3d7ff)
Target: x86_64-unknown-unknown
Thread model: posix
InstalledDir: /root/ispc/build/stage2/bin
 (in-process)
 "/root/ispc/build/stage2/bin/clang-18" -cc1 -triple x86_64-unknown-unknown -emit-llvm-bc -emit-llvm-uselists -disable-free -clear-ast-before-backend -disable-llvm-verifier -discard-value-names -main-file-name dispatch.c -mrelocation-model pic -pic-level 2 -fhalf-no-semantic-interposition -mframe-pointer=all -fmath-errno -ffp-contract=on -fno-rounding-math -mconstructor-aliases -funwind-tables=2 -target-cpu core2 -tune-cpu generic -debugger-tuning=gdb -fdebug-compilation-dir=/root/ispc -v -fcoverage-compilation-dir=/root/ispc -resource-dir /root/ispc/build/stage2/lib/clang/18 -D MACOS -O2 -ferror-limit 19 -fgnuc-version=4.2.1 -fskip-odr-check-in-gmf -fcolor-diagnostics -vectorize-loops -vectorize-slp -faddrsig -D__GCC_HAVE_DWARF2_CFI_ASM=1 -o /root/ispc/build/build-ispc-stage2/src/ispc-stage2-build/bin/../share/ispc/builtins_dispatch_macos.bc -x c builtins/dispatch.c
clang -cc1 version 18.1.8 based upon LLVM 18.1.8 default target x86_64-unknown-linux-gnu
#include "..." search starts here:
#include <...> search starts here:
 /usr/local/include
 /root/ispc/build/stage2/lib/clang/18/include
 /usr/include
End of search list.
In file included from builtins/dispatch.c:24:
In file included from /root/ispc/build/stage2/lib/clang/18/include/stdint.h:52:
/usr/include/stdint.h:26:10: fatal error: 'bits/libc-header-start.h' file not found
   26 | #include <bits/libc-header-start.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

It happened after PR https://github.com/ispc/ispc/pull/3029 that changed triple to `x86_64-unknown-unknown`. Compilation with target `x86_64-linux-gnu` is fine:


```bash
root@a147a4aec370:~/ispc# /root/ispc/build/stage2/bin/clang++ -x c -DMACOS -fPIC --target=x86_64-linux-gnu -march=core2 -mtune=generic -O2 -emit-llvm builtins/dispatch.c -c -o /root/ispc/build/build-ispc-stage2/src/ispc-stage2-build/bin/../share/ispc/builtins_dispatch_macos.bc -v
clang version 18.1.8 (https://github.com/llvm/llvm-project.git 3b5b5c1ec4a3095ab096dd780e84d7ab81f3d7ff)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /root/ispc/build/stage2/bin
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/9
Selected GCC installation: /usr/lib/gcc/x86_64-linux-gnu/9
Candidate multilib: .;@m64
Selected multilib: .;@m64
 (in-process)
 "/root/ispc/build/stage2/bin/clang-18" -cc1 -triple x86_64-unknown-linux-gnu -emit-llvm-bc -emit-llvm-uselists -disable-free -clear-ast-before-backend -disable-llvm-verifier -discard-value-names -main-file-name dispatch.c -mrelocation-model pic -pic-level 2 -fhalf-no-semantic-interposition -mframe-pointer=none -fmath-errno -ffp-contract=on -fno-rounding-math -mconstructor-aliases -funwind-tables=2 -target-cpu core2 -tune-cpu generic -debugger-tuning=gdb -fdebug-compilation-dir=/root/ispc -v -fcoverage-compilation-dir=/root/ispc -resource-dir /root/ispc/build/stage2/lib/clang/18 -D MACOS -internal-isystem /root/ispc/build/stage2/lib/clang/18/include -internal-isystem /usr/local/include -internal-isystem /usr/lib/gcc/x86_64-linux-gnu/9/../../../../x86_64-linux-gnu/include -internal-externc-isystem /usr/include/x86_64-linux-gnu -internal-externc-isystem /include -internal-externc-isystem /usr/include -O2 -ferror-limit 19 -fgnuc-version=4.2.1 -fskip-odr-check-in-gmf -fcolor-diagnostics -vectorize-loops -vectorize-slp -faddrsig -D__GCC_HAVE_DWARF2_CFI_ASM=1 -o /root/ispc/build/build-ispc-stage2/src/ispc-stage2-build/bin/../share/ispc/builtins_dispatch_macos.bc -x c builtins/dispatch.c
clang -cc1 version 18.1.8 based upon LLVM 18.1.8 default target x86_64-unknown-linux-gnu
ignoring nonexistent directory "/usr/lib/gcc/x86_64-linux-gnu/9/../../../../x86_64-linux-gnu/include"
ignoring nonexistent directory "/include"
#include "..." search starts here:
#include <...> search starts here:
 /root/ispc/build/stage2/lib/clang/18/include
 /usr/local/include
 /usr/include/x86_64-linux-gnu
 /usr/include
End of search list.
```

--- a/builtins/dispatch.c
+++ b/builtins/dispatch.c
@@ -21,8 +21,6 @@
 #error "Either REGULAR or MACOS macro need to defined"
 #endif
 
-#include <stdint.h>
-
 void abort();
 
 static int __system_best_isa = -1;
@@ -60,7 +58,7 @@ static int __os_has_avx512_support() {
 // __get_system_isa should return a value corresponding to one of the
 // Target::ISA enumerant values that gives the most capable ISA that the
 // current system can run.
-static int32_t __get_system_isa() {
+static int __get_system_isa() {
     int info[4];
     __cpuid(info, 1);
 

